Use the C++ current RpcAgent pointer to eliminate the unnecessary argument passing from Python world

xush6528 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#32635 Use the C++ current RpcAgent pointer to eliminate the unnecessary argument passing from Python world**
* #32633 Make C++ RpcAgent::currentRPCAgent_ the source of truth of current RPC Agent

With the source of truth of current RPC agent moved to C++ world, there is no point of passing current RPC agent from Python world to C++ world.

Differential Revision: [D5703519](https://our.internmc.facebook.com/intern/diff/D5703519/)

xush6528 committed:
Use the C++ current RpcAgent pointer to eliminate the unnecessary argument passing from Python world

With the source of truth of current RPC agent moved to C++ world, there is no point of passing current RPC agent from Python world to C++ world.

Differential Revision: [D5703519](https://our.internmc.facebook.com/intern/diff/D5703519/)

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit d57d22f8:


**None of the build failures appear to be your fault**.

* **1/1** [broken upstream](https://dr.pytorch.org/master-timeline.html?count=1&sha1=6ad9e5c70d1f496b9eaca191ec18e26c106c6436&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35) at merge base 6ad9e5c7 since Jan 25
  <details>
  <summary>
  Please rebase on the <code>viable/strict</code> branch (expand for instructions)
  </summary>
  
  Since your merge base is older than `viable/strict`, run these commands:
  ```
  git fetch origin viable/strict
  git rebase viable/strict
  ```
  Check out the [recency history](https://dr.pytorch.org/master-viable-commits.html) of this "viable master" tracking branch.
  
  </details>
  


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=d57d22f8aec572b12cad65f87b2c8a77a6efa7b0).

### :construction: 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably **caused by upstream breakages**:
* [pytorch_macos_10_13_py3_test](https://dr.pytorch.org/build-details.html?build_id=91076446) from Jan 25 until Jan 25



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

mrshenli commented on torch/csrc/distributed/rpc/python_functions.cpp:
```diff
@@ -83,8 +82,9 @@ std::shared_ptr<FutureMessage> sendPythonRemoteCall(
   // set forceGradRecording to true as even if the args does not contain any
   // tensor, the return value might still contain tensors.
+  auto agent = RpcAgent::getDefaultRpcAgent();
   return torch::distributed::autograd::sendMessageWithAutograd(
```
should this instead take a `std::shared_ptr&`?

xush6528 replied:
I guess it want to save the trouble of checking nullptr.
If agent is a nullptr. I guess *agent should raise an exception?
I don't want to change about it in this PR, unless I have clear nullptr checking guideline in mind.

mrshenli commented on torch/csrc/distributed/rpc/python_functions.cpp:
```diff
@@ -137,12 +136,12 @@ std::shared_ptr<FutureMessage> pyRpcBuiltin(
+  auto agent = RpcAgent::getDefaultRpcAgent();
   return sendMessageWithAutograd(
-      agent, dst, std::move(*scriptCall).toMessage(), false, rf);
+      *agent, dst, std::move(*scriptCall).toMessage(), false, rf);
```
ditto

mrshenli commented on torch/csrc/distributed/rpc/python_functions.cpp:
```diff
@@ -162,45 +161,46 @@ PyRRef pyRemoteBuiltin(
+  auto agent = RpcAgent::getDefaultRpcAgent();
   return sendMessageWithAutograd(
-      agent,
+      *agent,
```
ditto

mrshenli commented on torch/csrc/distributed/rpc/python_functions.cpp:
```diff
@@ -162,45 +161,46 @@ PyRRef pyRemoteBuiltin(
+  auto agent = RpcAgent::getDefaultRpcAgent();
   auto fm = sendMessageWithAutograd(
-      agent, dst, std::move(*scriptRemoteCall).toMessage(), false, rf);
+      *agent, dst, std::move(*scriptRemoteCall).toMessage(), false, rf);
```
ditto

mrshenli commented on torch/csrc/distributed/rpc/python_functions.cpp:
```diff
@@ -162,45 +161,46 @@ PyRRef pyRemoteBuiltin(
   auto& ctx = RRefContext::getInstance();
   auto serializedPyObj =
       SerializedPyObj(std::move(pickledPythonUDF), std::move(tensors));
+  auto agent = RpcAgent::getDefaultRpcAgent();
```
is this agent used anywhere below in the new code?

xush6528 replied:
no. should remove it after changing sendPythonRemoteCall.

mrshenli reviewed (approved):
LGTM! Thanks!

xush6528 committed:
Update on "Use the C++ current RpcAgent pointer to eliminate the unnecessary argument passing from Python world"


With the source of truth of current RPC agent moved to C++ world, there is no point of passing current RPC agent from Python world to C++ world.

Differential Revision: [D5703519](https://our.internmc.facebook.com/intern/diff/D5703519/)

[ghstack-poisoned]

xush6528 committed:
Update on "Use the C++ current RpcAgent pointer to eliminate the unnecessary argument passing from Python world"


With the source of truth of current RPC agent moved to C++ world, there is no point of passing current RPC agent from Python world to C++ world.

Differential Revision: [D5703519](https://our.internmc.facebook.com/intern/diff/D5703519/)

[ghstack-poisoned]

xush6528 committed:
Update on "Use the C++ current RpcAgent pointer to eliminate the unnecessary argument passing from Python world"


With the source of truth of current RPC agent moved to C++ world, there is no point of passing current RPC agent from Python world to C++ world.

Differential Revision: [D5703519](https://our.internmc.facebook.com/intern/diff/D5703519/)

[ghstack-poisoned]
